### ISSUE TITLE
# fmt: off block is partially reformatted

### ISSUE THREAD

**Issue by ikamensh** [psf/black#3372]
**Describe the bug**

**To Reproduce**
Run black on this code, and observe a change that should not have happened. 

```python
def recv(
    self,
) -> (
    # fmt: off
        gym_old_venv_step_type     # old API step
        | gym_new_venv_step_type   # new API step
        | tuple[np.ndarray, dict]  # new API reset
        | np.ndarray               # old API reset
):  # fmt: on # noqa:E125
    ...
```

Reformatting happens on the line before # fmt: on, with comment moved to the maximum left position.

**Expected behavior**

No reformatting takes place inside `# fmt: off` block

**Environment**

Both local 22.10 and https://black.vercel.app/?version=main

## Additional issue

the `# fmt: off` itself gets moved, which is not what I'd like, but then it's maybe less clear what should have happened.

**Comment by JelleZijlstra** [comment]
I'm not sure this is supposed to work. `# fmt: off` is documented as working on blocks (https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html), not on every possible subexpression.

**Comment by felix-hilden** [comment]
I agree, although I'd be open to converting this into a feature request. I'm not convinced that implementing this would be worth it but we could discuss.

**Comment by Conchylicultor** [comment]
I'm also affect:

Before:

```python
values = _parse_flags(
    # fmt:off
    './program'
    ' --test_config.other_with_default_overitten=True',  # Comment moved
    # fmt:on
)
```

After:

```python
values = _parse_flags(
    # fmt:off
    './program'
    ' --test_config.other_with_default_overitten=True',
    # Comment moved
    # fmt:on
)
```

The `# Comment moved` is moved on the next line
